qcengine/programs/molpro.py

```diff
@@ -45,11 +45,20 @@ class MolproHarness(ProgramHarness):
         "M12HFC", "HJSWPBE", "HJSWPBEH", "TCSWPBE", "PBESOL"
     }
 
-    # Currently supported methods in QCEngine for Molpro
-    _scf_methods: Set[str] = {"HF", "RHF", "KS", "RKS"}
-    _post_hf_methods: Set[str] = {'MP2', 'CCSD', 'CCSD(T)'}
-    _supported_methods: Set[str] = {*_scf_methods, *_post_hf_methods}
-
+    # Different open-shell scenarios:
+    # - Restricted references
+    #       RHF-RMP2 (like RHF-UMP2 using Molpro's naming convention for CC methods)
+    #       RHF-UCCSD
+    #       RHF-UCCSD(T)
+    #       RHF-RCCSD
+    #       RHF-RCCSD(T)
+    # - Unrestricted references (Only supported up to UMP2, no CC support)
+    #       UHF-UMP2
+    # NOTE: Unrestricted SCF methods must be specified by using keyword reference
+    _hf_methods: Set[str] = {"HF", "RHF"}
+    _restricted_post_hf_methods = {"MP2", "CCSD", "CCSD(T)"}  # RMP2, RCCSD, RCCSD(T)}
```

Comment:
Any reason these do not have types?

Reply:
Nope, just blanked on that. I'll add that in.

Reply:
hmm, I've been considering adding extra info to `model.method` as a violation of the QCSchema contract.

* `tce-ccsd` in nwc trying to add routing info to method
* `df-mp2` once upon a time in p4 trying to add algorithm info to method
* `uccsd` trying to add reference info to method
* I thought I remembered a gamess example last week but can't find it now.

my solution has been that the qcng interface must be taught a new keyword -- `qc_module` or `reference`

Reply:
I think I'd be in support of needing to specify exactly what's in QCSchema for `model.method`. Adding new keywords would then make it more consistent between different ProgramHarnesses (i.e. I would need to find the potentially different keyword that controls density fitting in Molpro and Psi4 instead of allowing `df-mp2` to be a valid `model.method` for Molpro and not Psi4.